Mar-updates (#403)
zalando/zalando.github.io: _posts/2019-04-25-mar-updates.md

---
title: March Updates -  A new Kubernetes operator & more Cloud Native Apps. 
author: Hong Phuc Dang 
date: 2019-04-25
banner:
  image: mar-banner.jpg
---
### This is a recap of open source activities and development at Zalando in the month of March

## Project Highlights 

> A new operator is added to Zalando’s list of Cloud Native Applications.  [Es-Operator](https://github.com/zalando-incubator/es-operator) - an operator for running Elasticsearch in Kubernetes with focus on operational aspects, like safe draining and offering auto-scaling capabilities for Elasticsearch data nodes, rather than just abstracting manifest definitions.

Reviewer: Ensure these listings are consistent so that is Project Name and Link - Then description, instead of embedding the project inside the text, it makes it hard to identify what project it is being released and how to find it

Author: I don't understand, could you please help to change it. Thanks.

> A new operator is added to Zalando’s list of Cloud Native Applications.  [Es-Operator](https://github.com/zalando-incubator/es-operator) - an operator for running Elasticsearch in Kubernetes with focus on operational aspects, like safe draining and offering auto-scaling capabilities for Elasticsearch data nodes, rather than just abstracting manifest definitions.

Reviewer 2: please name it either ES-Operator or "Elasticsearch Operator"

> To make things even simpler for developers, we also released a new framework that helps to build Kubernetes operators in Python. [Kopf - Kubernetes Operator Pythonic Framework](https://github.com/zalando-incubator/kopf) - a  framework and a library to make Kubernetes operators development easier, just in few lines of Python code. The main goal is to bring the Domain-Driven Design to the infrastructure level, with Kubernetes being an orchestrator/database of the domain objects (custom resources), and the operators containing the domain logic (with no or minimal infrastructure logic).

Reviewer 3: 👍

> **Weekly Open Source Maintainer Day at Developer Productivity** - Pionioneer in open source development at Zalando, Developer Productivity has actively released free software and contributed to many upstream projects. Since the beginning of this year, the Cloud Infrastructure team leading by [Jannis Rake-Revelant](https://twitter.com/jannis_r), reserved every Tuesday for open source maintaining work - which they called ‘Maintainer Tuesday’. It is a part of Zalando commitment to open source community. 

Reviewer: this is internal - we have channels for this, so we should avoid mixing what is useful for internal and external audiences

Author: I think it is relevant to external. Google broadcast to the world that they give engineers 20% to work on their own ideas. It is to promote the working environment at Zalando.

Reviewer 4: I agree, it's worth a small mention that a team dedicates one full day to OSS.

Reviewer 5: Should it be "pioneer" or is "pionioneer" intentional?
```suggestion
**Weekly Open Source Maintainer Day at Developer Productivity** - Pioneer in open source development at Zalando, Developer Productivity has actively released free software and contributed to many upstream projects. Since the beginning of this year, the Cloud Infrastructure team leading by [Jannis Rake-Revelant](https://twitter.com/jannis_r), reserved every Tuesday for open source maintaining work - which they called ‘Maintainer Tuesday’. It is a part of Zalando commitment to open source community. 
```

Reviewer: Fair arguments, then re-write with an external audience in mind:

1. Explain what "developer productivity" means, ex. projects they maintain etc. so there is context for people not aware of internal structures
2. provide more information - how many developers are doing this every week, maybe there are already tangible examples of how this benefits external users of our projects (ex: due to this we able to release X and invest Y number of hours into maturing Z) 

Especially when we use claims such as "commitment to open source" and "pioneering" we should back up those claims with tangible examples - which I know we can for sure.

Author: Thank you for the feedback. I'll see if I can add any additional info. However this is a blog article  - not a scientific paper - I think it is also fine if we don't have evidence for every sentence :)

Reviewer: No need for scientific level info... - just provide a context so external readers understand what this means

> **Weekly Open Source Maintainer Day at Developer Productivity** - Pioneering in open source development at Zalando, Platform Infrastructure department has actively released free software and contributed to many upstream projects. Since the beginning of this year, the Cloud Infrastructure team led by [Jannis Rake-Revelant](https://twitter.com/jannis_r), reserved every Tuesday for open source maintaining work - which they called ‘Maintainer Tuesday’. It is part of Zalando's commitment to open source community. 

Reviewer: ```suggestion
**Dedicated Open Source Time In The Zalando Cloud Infrastructure Team** The engineering team led by [Jannis Rake-Revelant](https://twitter.com/jannis_r), who is responsible for some our most popular open source projects have, since the beginning of the year, dedicated 20% of their time to ensure their open source projects are actively maintained and improved.  As a company we believe it is important to take long term responsibility and show commitment to the open source community which we benefit from every day.
```

> **Weekly Open Source Maintainer Day at Developer Productivity** - Pionioneer in open source development at Zalando, Platform Infrastructure department has actively released free software and contributed to many upstream projects. Since the beginning of this year, the Cloud Infrastructure team leading by [Jannis Rake-Revelant](https://twitter.com/jannis_r), reserved every Tuesday for open source maintaining work - which they called ‘Maintainer Tuesday’. It is a part of Zalando commitment to open source community. 

Reviewer 4: "Pioneering"

Reviewer 4: "team led by"

Reviewer 4: "It is part of Zalando's commitment to open source"

Author: @hjacobs fixed

> **Weekly Open Source Maintainer Day at Developer Productivity** - Pionioneer in open source development at Zalando, Platform Infrastructure department has actively released free software and contributed to many upstream projects. Since the beginning of this year, the Cloud Infrastructure team leading by [Jannis Rake-Revelant](https://twitter.com/jannis_r), reserved every Tuesday for open source maintaining work - which they called ‘Maintainer Tuesday’. It is a part of Zalando commitment to open source community. 

Reviewer 4: "Developer Productivity" -> "Platform Infrastructure" (like in the text)

 
## Zalando Around The World 
Meet and connect with Zalando representatives at tech events around the world: 

[GAIA Conference](https://www.gaia.fish), Göteborg, Apr 9: Mikio Braun - our AI expert - gave a keynote on Putting Data Science into Production. Check out his talk below: 

<iframe width="800" height="590" src="https://www.youtube.com/embed/jePTtEFBgLI" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
--

[Strata Conference](https://conferences.oreilly.com/strata/strata-eu), London, Apr 29 - May 2: Dirk Petzoldt - Head of Engineering will share how Zalando handles big data in our online marketing platform. [More details](https://conferences.oreilly.com/strata/strata-eu/public/schedule/detail/74071)

[Coding Serbia Conference](https://codingserbia.com), Novi Sad, May 15 - 17: [Luis Mineiro](https://de.linkedin.com/in/lmineiro) - Senior Site Reliability Engineer,  will explain how we set up monitoring and alerting at Zalando and go over the basic concepts of Distributed Tracing and OpenTracing. 

> 

Reviewer: ```suggestion
[GitHub Satellite](https://githubsatellite.com/), Berlin, May 22 - 23: [Per Ploug](https://www.linkedin.com/in/per-ploug-krogslund/) - Open Source Manager,  will talk about Open Source and security and try to answer the question: who is actually responsible for the security of open source dependencies?. 

```
